qiskit/primitives/base_estimator.py

```diff
@@ -197,28 +228,28 @@ def parameters(self) -> tuple[ParameterView, ...]:
         """
         return self._parameters
 
-    @abstractmethod
+    @deprecate_arguments({"circuit_indices": "circuits", "observable_indices": "observables"})
     def __call__(
         self,
-        circuit_indices: Sequence[int],
-        observable_indices: Sequence[int],
-        parameter_values: Sequence[Sequence[float]],
+        circuits: Sequence[int | QuantumCircuit],
+        observables: Sequence[int | SparsePauliOp],
+        parameter_values: Sequence[Sequence[float]] | None = None,
```

Comment:
It feels like the types between `tuple`, `Iterable` and `Sequence` are not quite consistent in the different functions. Is this intended, or should the be the same (e.g. `parameters` are always `Iterable`s and `circuits` always `tuple`s)?

Reply:
Basically, Sequence should be fine... Sorry for inconsistency. I'll fix it.

Reply:
Sorry...on second thought on this point I didn't know the reason. The commit 8a70ff0 in #7723 introduced it, but I'm not sure.